File: app/src/main/java/br/fmu/bioreino/dao/CursosDAO.java
package br.fmu.bioreino.dao;

import java.util.ArrayList;

import br.fmu.bioreino.model.Aula;
import br.fmu.bioreino.model.Categoria;
import br.fmu.bioreino.model.Curso;

public class CursosDAO {

    /// Lista de todos os cursos que formos adicionar, simula um banco de dados
    private static final ArrayList<Curso> cursos = new ArrayList<>();
    public static Curso ultimoCurso;

    /// Retorna a lista de todos os cursos
    public static ArrayList<Curso> getCursos() {
        return cursos;
    }

    /// Adiciona um curso à lista estática
    public static void adicionaCurso(Curso curso) {
        cursos.add(curso);
    }

    /// Liste de categorias
    private static final ArrayList<Categoria> categorias = new ArrayList<>();

    public static void adicionaCategoria(Categoria categoria) {
        categorias.add(categoria);
    }

    public static ArrayList<Categoria> getCategorias() {
        return categorias;
    }

    /// Lista de aulas
    public static void setAulaVistaTrue(int aulaIndex, Curso curso) {
        cursos.get(cursos.indexOf(curso)).getAulas().get(aulaIndex).setAulaVista(true);
    }

    public static void setUltimaAula(int aulaIndex, Curso cursoInformado) {
        Curso curso = cursos.get(cursos.indexOf(cursoInformado));
        Aula aula = curso.getAulas().get(aulaIndex);
        curso.setUltimaAula(aula);
    }

    public static Aula getUltimaAula(Curso curso) {
        return cursos.get(cursos.indexOf(curso)).getUltimaAula();
    }


}

File: app/src/main/java/br/fmu/bioreino/model/dao/CursosDAO.kt
package br.fmu.bioreino.model.dao

import br.fmu.bioreino.model.entities.Aula
import br.fmu.bioreino.model.entities.Categoria
import br.fmu.bioreino.model.entities.Curso

object CursosDAO {
    /// Retorna a lista de todos os cursos
    /// Lista de todos os cursos que formos adicionar, simula um banco de dados
    @JvmStatic
    val cursos = ArrayList<Curso>()
    @JvmField
    var ultimoCurso: Curso? = null

    /// Adiciona um curso à lista estática
    fun adicionaCurso(curso: Curso) {
        cursos.add(curso)
    }

    /// Liste de categorias
    val categorias = ArrayList<Categoria>()
    fun adicionaCategoria(categoria: Categoria) {
        categorias.add(categoria)
    }

    /// Lista de aulas
    @JvmStatic
    fun setAulaVistaTrue(aulaIndex: Int, curso: Curso) {
        cursos[cursos.indexOf(curso)].aulas[aulaIndex].setAulaVista(true)
    }

    @JvmStatic
    fun setUltimaAula(aulaIndex: Int, cursoInformado: Curso) {
        val curso = cursos[cursos.indexOf(cursoInformado)]
        val aula = curso.aulas[aulaIndex]
        curso.ultimaAula = aula
    }

    fun getUltimaAula(curso: Curso): Aula? {
        return cursos[cursos.indexOf(curso)].ultimaAula
    }
}